it/endpoints/PostProcessReviewISpec.scala

```diff
@@ -30,47 +31,60 @@ import utils.Constants._
 
 class PostProcessReviewISpec extends IntegrationSpec {
 
-  "Calling the approval2iReviewComplete POST endpoint with a valid payload" when {
+  val statusChangeInfo: ApprovalProcessStatusChange = ApprovalProcessStatusChange("user id", "user name", StatusWithDesignerForUpdate)
+
+  val statusChangeWithDesignerJson: JsValue = Json.toJson(statusChangeInfo)
+
+  "Calling the approval2iReviewComplete POST endpoint with a valid payload and all pages reviewed" when {
+
+    val pageUrl = "/feeling-bad"
 
     def populateDatabase(processToSave: JsValue): String = {
-      lazy val request = buildRequest("/external-guidance/approval")
+      lazy val request = buildRequest("/external-guidance/approval/2i-review")
 
       val result = await(request.post(processToSave))
       val json = result.body[JsValue].as[JsObject]
-      (json \ "id").as[String]
-    }
+      val id = (json \ "id").as[String]
+      lazy val pageUpdateRequest = buildRequest(s"/external-guidance/approval/$id/2i-page-review$pageUrl")
+      val content = ApprovalProcessPageReview("1", pageUrl, Some("Yes"), ReviewCompleteStatus, Some("A basic comment"), LocalDateTime.now(), Some("User1"))
+      await(pageUpdateRequest.post(Json.toJson(content)))
 
+      id
+    }
     val processToSave: JsValue = simpleValidProcess
-    lazy val id = populateDatabase(processToSave)
-    lazy val request = buildRequest(s"/external-guidance/approval/$id/2i-review")
 
-    "the new status is submitted for 2i Review" should {
+    "the requested status update is With DesignerForUpdate" should {
+      lazy val id = populateDatabase(processToSave)
+      lazy val request = buildRequest(s"/external-guidance/approval/$id/2i-review")
 
-      lazy val response: WSResponse = {
-        AuditStub.audit()
-        await(request.post(statusChangeJson))
-      }
-
-      "return a NO_CONTENT status code" in {
-        response.status shouldBe NO_CONTENT
-      }
+      "set the new status to With Designer For Update" should {
 
-      "set the status to SubmittedFor2iReview" in {
-        lazy val request = buildRequest(s"/external-guidance/approval")
         lazy val response: WSResponse = {
           AuditStub.audit()
-          await(request.get())
+          await(request.post(statusChangeWithDesignerJson))
         }
-        val list: List[ApprovalProcessSummary] = response.body[JsValue].as[List[ApprovalProcessSummary]]
-        val updatedEntry = list.find(p => p.id == id)
-        updatedEntry shouldBe 'defined
-        updatedEntry.get.status shouldBe StatusSubmittedFor2iReview
-      }
 
-    }
+        "return a NO_CONTENT status code" in {
+          response.status shouldBe NO_CONTENT
+        }
 
-    "the new status is published" should {
+        "set the status to SubmittedFor2iReview" in {
```

Comment:
Is this description correct? The test seems to be checking the process's status has been set to with designer for update.

Reply:
have updated